community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/notifications/LogicalPlanNotifications.scala

```diff
@@ -17,19 +17,25 @@
  * You should have received a copy of the GNU General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
-package org.neo4j.cypher.internal.compatibility.v3_4
+package org.neo4j.cypher.internal.compatibility.v3_4.notifications
 
 import org.neo4j.cypher.internal.compiler.v3_4.CypherCompilerConfiguration
 import org.neo4j.cypher.internal.frontend.v3_4.notification.InternalNotification
 import org.neo4j.cypher.internal.planner.v3_4.spi.PlanContext
 import org.neo4j.cypher.internal.v3_4.logical.plans.LogicalPlan
 
+trait NotificationChecker {
+  def apply(plan: LogicalPlan): TraversableOnce[InternalNotification]
```

Comment:
Option would not work because of the new `NotificationChecker` we added, that would have more than one Notification. `TraversableOnce` is just a very high level interface that allows us to do `flatMap`, which is all we want here. Then the implementations can choose themselves if `Seq` or `Iterator` or whatnot is the best choice.